pypeit/core/flat.py

```diff
@@ -493,24 +508,28 @@ def flatfield(sciframe, flatframe, bpix, illum_flat=None, snframe=None, varframe
 
     # New image
     retframe = np.zeros_like(sciframe)
-    w = np.where(final_flat > 0.0)
-    retframe[w] = sciframe[w]/final_flat[w]
-    if w[0].size != final_flat.size:
-        ww = np.where(final_flat <= 0.0)
-        bpix[ww] = 1.0
-    # Variance?
-    if varframe is not None:
-        # This is risky -- Be sure your flat is well behaved!!
-        retvar = np.zeros_like(sciframe)
-        retvar[w] = varframe[w]/final_flat[w]**2
-        return retframe, retvar
-    # Error image
-    if snframe is None:
+#    w = np.where(final_flat > 0.0)
+    # TODO: use utils.inverse() instead?
+    indx = final_flat > 0.0
+    retframe[indx] = sciframe[indx]/final_flat[indx]
+    if bpm is not None and np.sum(indx) != final_flat.size:
+        bpm[np.logical_not(indx)] = 1
```

Comment:
Let's talk through this change. I'm of the opinion that the bpm should be immutable, and something defined by the user. If the flat field code is finding some pixels are bad, I'd be inclined to just set those bad flat field pixels to one, and simply do nothing to the image. Then if the reduced has bad pixels, the user can update the immutable bpm.

Reply:
As per our discussion over zoom, actually the bpm is not passed now.  I just return a mask based on when final_flat is <= 0.